__device-tests__/gutenberg-editor-sanity-test-4-visual.test.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+const { toggleOrientation, selectTextFromTextInput, setClipboard } = e2eUtils;
+import { takeScreenshot } from './utils';
+
+describe( 'Gutenberg Editor - Test Suite 4', () => {
+	it( 'Spacer block - Spacer in horizontal layout works as expected', async () => {
+		await editorPage.addNewBlock( blockNames.spacer );
+
+		await toggleOrientation( editorPage.driver );
+
+		// Wait for the device to finish rotating
+		await editorPage.driver.sleep( 3000 );
+
+		// Visual test check for selected state
+		let screenshot = await takeScreenshot();
+		expect( screenshot ).toMatchImageSnapshot();
+
+		const titleElement = await editorPage.getTitleElement( {
+			autoscroll: true,
+		} );
+		await titleElement.click();
+
+		await editorPage.dismissKeyboard();
+
+		// Visual test check for unselected state
+		screenshot = await takeScreenshot();
+		expect( screenshot ).toMatchImageSnapshot();
+
+		await toggleOrientation( editorPage.driver );
+
+		// Wait for the device to finish rotating
+		await editorPage.driver.sleep( 3000 );
+
+		const spaceBlock = await editorPage.getBlockAtPosition(
+			blockNames.spacer
+		);
+		await spaceBlock.click();
+		await editorPage.removeBlock();
+	} );
+
+	describe( 'Buttons block', () => {
+		const buttonCustomColors = `<!-- wp:buttons -->
+<div class="wp-block-buttons"><!-- wp:button {"style":{"color":{"background":"#b1005b","text":"#ffe6f2"}}} -->
+<div class="wp-block-button"><a class="wp-block-button__link has-text-color has-background wp-element-button" style="color:#ffe6f2;background-color:#b1005b">Button</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`;
+
+		it( 'Render custom text and background color', async () => {
+			await editorPage.setHtmlContent( buttonCustomColors );
+
+			const buttonsBlock = await editorPage.getBlockAtPosition(
+				blockNames.buttons
+			);
+			expect( buttonsBlock ).toBeTruthy();
```

Comment:
Are the `toBeTruthy` assertions in this file testing for something specific that is separate from `toMatchImageSnapshot`?

Reply:
Since we are adding this block from the HTML view and then toggling back, to have a way of waiting for elements to render correctly before taking a screenshot I used the `toBeTruthy` assertion to make sure the block is there before the next step.

Reply:
Ah, I see. The `getBlockAtPosition` is the key line awaiting the required render. The assertion is an attempt at communicating the intent. 👍🏻